File: 1984-maximum-distance-between-a-pair-of-values/maximum-distance-between-a-pair-of-values.java
class Solution {
    public int maxDistance(int[] nums1, int[] nums2) {
        int i = 0;
        int j = 0;
        int n = nums1.length;
        int m = nums2.length;
        int ans = 0;
        while(i < n && j < m) {
            if(nums1[i] > nums2[j]) {
                i++;
                continue;
            }

            ans = Math.max(ans, j - i);
            j++;
        }
        return ans;
    }
}

File: 1984-maximum-distance-between-a-pair-of-values/maximum-distance-between-a-pair-of-values.cpp
class Solution {
public:
    int maxDistance(vector<int>& nums1, vector<int>& nums2) {
        int ans = 0;
        int n = nums1.size();
        int m = nums2.size();
        int i = 0;

        for(int j = 0; j < m; j++) {
            
            while(i < n && nums1[i] > nums2[j]) {
                i++;
            }

            if(i == n) {
                break;
            }
            if(i <= j) {
                ans = max(ans, (j-i));
            }
        }

        return ans;
    }
};